source/ddbus/thin.d

```diff
@@ -106,6 +140,35 @@ struct ObjectPath {
     return objPath.representation.splitter('/').drop(1).all!(a => a.length
         && a.all!(c => c.isAlphaNum || c == '_'));
   }
+
+  /// Does an unsafe assignment to an ObjectPath.
+  static ObjectPath assumePath(string path) pure @nogc nothrow @safe {
+    ObjectPath ret;
+    ret._value = path;
+    return ret;
+  }
+}
+
+/// Serves as typesafe alias. Instances should be created using busName instead of casting.
+/// It prevents accidental usage of bus names in other string parameter fields and makes the API clearer.
+enum BusName : string {
```

Comment:
It is not really correct to use an ˋenumˋ type this way. Maybe use ˋstd.typecons.Typedefˋ instead?

Reply:
yes [as per spec](https://dlang.org/spec/enum.html#named_enums) (point 5) it allows the base type to not be assigned without casting, but allows implicit casting to the base type. This makes it a perfect typesafe alias. Typedef is actually a separate type which is a lot more awkward to use and uses templates. This is a very lightweight and imo a better solution than Typedef

Reply:
Seems technically correct, but semantically an enum is defined as a type that explicitly states it's possible values. Though it seems the spec currently allows assignment of a value not specified in the `enum` (I'm sure it used to prohibit that in some earlier version, probably with an exception for flag enums).

Reply:
All dmd versions since 2.060.0 allow
```
enum A : int { init = 0 }

void main()
{
  A foo = cast(A) 4;
    
  int x = foo;
}
```

It's an idiom I have used for a while and it is really useful and lightweight on the compiler (no templates)

LDC also implements it like that too and I think it's a far better way than overloading 20 operators to achieve similar things